hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatProtobuf.java

```diff
@@ -172,13 +172,72 @@ public LoaderContext getLoaderContext() {
       return ctx;
     }
 
+    private static class DigestThread extends Thread {
```

Comment:
Javadoc explaining the overall idea. Something like:
Thread to compute the MD5 of a file as this can be in parallel while loading the image without interfering much.

Actually, this might be a good util to have in MD5FileUtils.

Reply:
Thank @goiri   for reviewing this patch. Its purpose is  computing file md5 in parallel with image loading. So I think it might not be a good util to have in MD5FileUtils. Please correct me if I am wrong. Thanks.